From 82971b5e74509561afab748c3cedf9bab909d76b Mon Sep 17 00:00:00 2001
From: Simon Pena <spena@igalia.com>
Date: Fri, 25 Apr 2025 06:28:05 -0700
Subject: [PATCH 5/7] [Android] Declare namespace when building WebDriver as a
 library https://bugs.webkit.org/show_bug.cgi?id=292008

Reviewed by Michael Catanzaro and Adrian Perez de Castro.

When building WebDriver as a library on Android, ensure the WebKit
namespace is declared.

* Source/WebDriver/WebDriverMain.cpp: moved entry point
  WebDriverProcessMain into WebKit namespace.

Canonical link: https://commits.webkit.org/294105@main
---
 Source/WebDriver/WebDriverMain.cpp | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/Source/WebDriver/WebDriverMain.cpp b/Source/WebDriver/WebDriverMain.cpp
index df15263697a4..dd3b0286442d 100644
--- a/Source/WebDriver/WebDriverMain.cpp
+++ b/Source/WebDriver/WebDriverMain.cpp
@@ -36,8 +36,9 @@
 // https://github.com/Igalia/wpe-android/blob/b918e3f8b86eda406436cb251c2e7b10a529008c/wpeview/src/main/cpp/Service/EntryPoint.cpp#L55
 //
 #if OS(ANDROID)
+namespace WebKit {
 __attribute__((visibility("default")))
-int WebKit::WebDriverProcessMain(int argc, char** argv)
+int WebDriverProcessMain(int argc, char** argv)
 #else
 int main(int argc, char** argv)
 #endif
@@ -49,3 +50,6 @@ int main(int argc, char** argv)
     WebDriver::WebDriverService service;
     return service.run(argc, argv);
 }
+#if OS(ANDROID)
+}
+#endif
-- 
2.43.0

